src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs

```diff
@@ -0,0 +1,152 @@
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Telemetry
+{
+    public class TelemetryCommandExtension : AgentService, IWorkerCommandExtension
+    {
+        private string _area;
+        private string _feature;
+        public HostTypes SupportedHostTypes => HostTypes.Build;
+
+        public void ProcessCommand(IExecutionContext context, Command command)
+        {
+            if (string.Equals(command.Event, WellKnownEventTrackCommand.Publish, StringComparison.OrdinalIgnoreCase))
+            {
+                ProcessPublishTelemetryCommand(context, command.Properties, command.Data);
+            }
+            else
+            {
+                throw new Exception(StringUtil.Loc("TelemetryCommandNotFound", command.Event));
+            }
+        }
+
+        public Type ExtensionType
+        {
+            get
+            {
+                return typeof(IWorkerCommandExtension);
+            }
+        }
+
+        public string CommandArea
+        {
+            get
+            {
+                return "telemetry";
+            }
+        }
+
+        private void ProcessPublishTelemetryCommand(IExecutionContext context, Dictionary<string, string> eventProperties, string data)
+        {
+            ArgUtil.NotNull(context, nameof(context));
+
+            LoadTelemetryInputs(eventProperties);
+
+            var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
+            var vssConnection = WorkerUtilies.GetVssConnection(context);
+            ciService.Initialize(vssConnection);
+
+            var commandContext = HostContext.CreateService<IAsyncCommandContext>();
+            commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
+            commandContext.Task = ciService.PublishEventsAsync(new CustomerIntelligenceEvent[] { PopulateCustomerIntelligenceData(context, data) });
+            context.AsyncCommands.Add(commandContext);
+        }
+
+        private void LoadTelemetryInputs(Dictionary<string, string> eventProperties)
+        {
+            eventProperties.TryGetValue(WellKnownEventTrackProperties.Area, out _area);
+            if (string.IsNullOrEmpty(_area))
+            {
+                throw new ArgumentException(StringUtil.Loc("ArgumentNeeded", "Area"));
+            }
+
+            eventProperties.TryGetValue(WellKnownEventTrackProperties.Feature, out _feature);
+            if (string.IsNullOrEmpty(_feature))
+            {
+                throw new ArgumentException(StringUtil.Loc("ArgumentNeeded", "Feature"));
+            }
+        }
+
+        private CustomerIntelligenceEvent PopulateCustomerIntelligenceData(IExecutionContext context, string data)
+        {
+            if (string.IsNullOrEmpty(data))
+            {
+                throw new ArgumentException(StringUtil.Loc("ArgumentNeeded", "EventTrackerData"));
+            }
+
+            // Expected format is key1=Value1;key2=Value2;Key3=Value3
```

Comment:
Thought about it. since mostly our commands are following this pattern `key=value;`, I tried to maintain the same here. 
We can also serialize and deserialize however our commands dont support multi-line. so one has to serialize into compact mode and doesnt sound great to me.
telemetry is something to track minute data points and its not a logger where someone should send big objects, data.